http-clients/netty-nio-client/src/main/java/software/amazon/awssdk/http/nio/netty/internal/utils/NettyUtils.java

```diff
@@ -130,13 +130,19 @@ private NettyUtils() {
      *
      * @param eventExecutor Executor to run task in.
      * @param runnable Task to run.
+     *
+     * @return The {@code Future} from from the executor.
      */
-    public static void doInEventLoop(EventExecutor eventExecutor, Runnable runnable) {
+    public static Future<?> doInEventLoop(EventExecutor eventExecutor, Runnable runnable) {
         if (eventExecutor.inEventLoop()) {
-            runnable.run();
-        } else {
-            eventExecutor.submit(runnable);
+            try {
+                runnable.run();
+                return eventExecutor.newSucceededFuture(null);
+            } catch (Throwable t) {
+                return eventExecutor.newFailedFuture(t);
+            }
         }
+        return eventExecutor.submit(runnable);
```

Comment:
Do we need to handle the case that doInEventLoop can fail everywhere that we're currently using doInEventLoop?

Reply:
See comment here: https://github.com/aws/aws-sdk-java-v2/pull/2739#discussion_r719582253. Would be a less invasive change